docs/concepts/models/python_models.md

```diff
@@ -90,7 +90,40 @@ Optional pre/post-statements allow you to execute SQL commands before and after
 
 For example, pre/post-statements might modify settings or create indexes. However, be careful not to run any statement that could conflict with the execution of another statement if models run concurrently, such as creating a physical table.
 
-Pre- and post-statements are issued with the SQLMesh [`fetchdf` method](../../reference/cli.md#fetchdf) [described above](#execution-context).
+You can set the `pre_statements` and `post_statements` arguments to a list of SQL strings and/or SQLGlot expressions to define the model's pre/post-statements.
+
+``` python linenums="1" hl_lines="6-7"
+def execute(
+    context: ExecutionContext,
+    start: datetime,
+    end: datetime,
+    execution_time: datetime,
+    pre_statements=["SET GLOBAL parameter = 'value';"],
+    post_statements=["@CREATE_INDEX('example.pre_post_statements', id)"],
+    **kwargs: t.Any,
+) -> pd.DataFrame:
+
+    return pd.DataFrame([
+        {"id": 1, "name": "name"}
+    ])
+
+```
+
+Within these pre/post-statements, you can also use SQLMesh or Jinja macros. For example, a macro that creates an index based on the runtime stage:
```

Comment:
Replace with:

The previous example's `post_statements` called user-defined SQLMesh macro `@CREATE_INDEX('example.pre_post_statements', id)`.

We could define the `CREATE_INDEX` macro in the project's `macros` directory like this. The macro creates a table index on a single column, conditional on the [runtime stage](../../macros/macro_variables.md#runtime-variables) being `creating` (table creation time).